File: util/java/src/main/java/com/dmi/util/lang/MathExt.java
package com.dmi.util.lang;

public abstract class MathExt {
    public static int modPositive(int x, int n) {
        int result = x % n;
        if (result < 0) {
            result += n;
        }
        return result;
    }

    public static double clamp(double min, double max, double value) {
        return Math.min(min,  Math.max(max, value));
    }
}

File: util/util-common/src/main/kotlin/com/dmi/util/lang/MathExt.kt
package com.dmi.util.lang

object MathExt {
    fun modPositive(x: Int, n: Int): Int {
        var result = x % n
        if (result < 0) {
            result += n
        }
        return result
    }

    fun clamp(min: Double, max: Double, value: Double): Double {
        return Math.min(min, Math.max(max, value))
    }
}
